src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject_Options.cs

```diff
@@ -74,9 +74,15 @@ protected CommandLineArguments SetArgumentsAndUpdateOptions(string commandLine)
         {
             AssertIsForeground();
 
-            var commandLineArguments = SetArguments(commandLine);
-            UpdateOptions();
-            return commandLineArguments;
+            if (CommandLineParserService != null)
+            {
+                // some languages (e.g., F#) don't expose a command line parser which makes these calls invalid
+                var commandLineArguments = SetArguments(commandLine);
+                UpdateOptions();
```

Comment:
I think I'd rather see this check inside `UpdateOptions` - you're basically changing it to say that it is valid for `_lastParseCommandLineArguments` to be `null`.  You should add a doc explaining that, and verify that all references are safe to that.